### ISSUE TITLE
Terraform 0.13.1 crash during plan

### ISSUE THREAD

**Issue by sdebruyn** [hashicorp/terraform#26040]
### Terraform Version
Terraform 0.13.1

### Terraform Configuration Files
Sorry, infra is big and private, can share only specific bits but I don't know what caused it.

### Debug Output
```
------------------------------------------------------------------------
panic: not a collection type

goroutine 2415 [running]:
github.com/zclconf/go-cty/cty.Type.ElementType(0x2ce50a0, 0xc000fa0538, 0xc0008e18f8, 0x8)
	/home/circleci/project/project/vendor/github.com/zclconf/go-cty/cty/collection.go:25 +0xba
github.com/hashicorp/terraform/terraform.createEmptyBlocks(0xc000d07650, 0x2ce50a0, 0xc000fa0550, 0x2300040, 0xc001c570b0, 0x2ce50a0, 0xc000fa0688, 0x2300040, 0xc001ed1fb0)
	/home/circleci/project/project/terraform/eval_read_data_plan.go:242 +0x202
github.com/hashicorp/terraform/terraform.dataObjectHasChanges(0xc000d07650, 0x2ce50a0, 0xc000fa0550, 0x2300040, 0xc001c570b0, 0x2ce50a0, 0xc000fa0688, 0x2300040, 0xc001ed1fb0, 0x0, ...)
	/home/circleci/project/project/terraform/eval_read_data_plan.go:211 +0x19e
github.com/hashicorp/terraform/terraform.(*evalReadDataPlan).Eval(0xc002b4a000, 0x2d26860, 0xc000e14d00, 0x1, 0x1, 0x0, 0x0)
	/home/circleci/project/project/terraform/eval_read_data_plan.go:107 +0x1238
github.com/hashicorp/terraform/terraform.EvalRaw(0x2c7d420, 0xc002b4a000, 0x2d26860, 0xc000e14d00, 0x0, 0x0, 0x0, 0x0)
	/home/circleci/project/project/terraform/eval.go:49 +0xc7
github.com/hashicorp/terraform/terraform.(*EvalSequence).Eval(0xc000c13160, 0x2d26860, 0xc000e14d00, 0x1, 0x1, 0x6, 0xc0004a4770)
	/home/circleci/project/project/terraform/eval_sequence.go:20 +0xfd
github.com/hashicorp/terraform/terraform.EvalRaw(0x2c7ce80, 0xc000c13160, 0x2d26860, 0xc000e14d00, 0x22b4bc0, 0x3fa2622, 0x2214980, 0xc000a8c780)
	/home/circleci/project/project/terraform/eval.go:49 +0xc7
github.com/hashicorp/terraform/terraform.Eval(0x2c7ce80, 0xc000c13160, 0x2d26860, 0xc000e14d00, 0xc000c13160, 0x2c7ce80, 0xc000c13160, 0x0)
	/home/circleci/project/project/terraform/eval.go:35 +0x4d
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x2645300, 0xc00164bd10, 0x0, 0x0, 0x0)
	/home/circleci/project/project/terraform/graph.go:73 +0xc77
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc0004a4180, 0x2645300, 0xc00164bd10, 0xc00123e540)
	/home/circleci/project/project/dag/walk.go:387 +0x357
created by github.com/hashicorp/terraform/dag.(*Walker).Update
	/home/circleci/project/project/dag/walk.go:309 +0x11d7
```

### Crash Output
No access, was on Azure DevOps

### Expected Behavior
Plan works

### Actual Behavior
Plan crashes

### Steps to Reproduce
1. `terraform init`
2. `terraform plan`

### Additional Context
Not the first plan

**Comment by sdebruyn** [comment]
This does not crash with Terraform 0.13.0

**Comment by nexxai** [comment]
Was resolved in https://github.com/hashicorp/terraform/pull/26028 and will be fixed in v0.13.2

**Comment by alisdair** [comment]
Closing as a duplicate of #26011, which as @nexxai points out will be fixed by the next release. There are some possible workarounds reported by other users in the linked issue. Thanks for reporting!
